### PR TITLE
Add a state scripts support.

### PR DESCRIPTION
**Pull request by pasinskim**

### CODE REVIEW

File: awriter/writer.go [hunk 1]
```diff
@@ -211,6 +228,13 @@ func writeHeader(tw *tar.Writer, devices []string, name string,
 		return errors.New("writer: can not store header-info")
 	}
 
+	// write scripts
+	if scr != nil {
+		if err := writeScripts(tw, scr); err != nil {
+			return err
+		}
+	}
+
```
Comment (kacf) [inline awriter/writer.go]: Scripts are not valid in artifact v1. Does this need to be version protected?

File: artifacts.go [hunk 2]
```diff
@@ -32,6 +32,39 @@ import (
 // Version of the mender-artifact CLI tool
 var Version = "unknown"
 
+func version(c *cli.Context) int {
+	version := c.Int("version")
+	// set version to 2 if artifact is signed and version
+	// is not set explicitly
+	if !c.IsSet("version") && len(c.String("key")) != 0 {
+		version = 2
+	}
```
Comment (kacf) [inline artifacts.go]: You need to do this for scripts too, right?

File: artifacts.go [hunk 3]
```diff
@@ -32,6 +32,39 @@ import (
...
+		}
+		return awriter.NewWriterSigned(f, artifact.NewSigner(privateKey)), nil
+	}
+	return awriter.NewWriter(f), nil
+}
+
+func scripts(c *cli.Context) (*artifact.Scripts, error) {
+	scr := artifact.Scripts{}
+	if len(c.StringSlice("script")) > 0 {
+		for _, script := range c.StringSlice("script") {
```
Comment (kacf) [inline artifacts.go]: You might be able to lose the if here. If the range is zero it should just skip over the loop.

File: artifact/scripter.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+var availableScriptType = map[string]bool{
+	"Idle":                true,
+	"Sync":                true,
+	"ArtifactDownload":    true,
+	"ArtifactPreinstall":  true,
+	"ArtifactInstall":     true,
+	"Reboot":              true,
+	"ArtifactPostinstall": true,
+	"ArtifactCommit":      true,
```
Comment (pasinskim) [inline artifact/scripter.go]: Thanks for spotting this. Probably I took a look at some subset `Order that states will usually be executed in:`. Anyway, fixed now and missing states are added.

File: awriter/writer.go (hunk 1 above)
Comment (pasinskim) [inline awriter/writer.go]: I don't think this needs to be protected here. The better place would be in `artifacts.go` where you are writing artifact reading command line parameters. I will add a check there.

File: artifacts.go (hunk 2 above)
Comment (pasinskim) [inline artifacts.go]: Yup. Will add a check for this.

File: artifacts.go (hunk 3 above)
Comment (pasinskim) [inline artifacts.go]: Fixing!

**Comment by pasinskim** [comment]
After rerunning build was green. The first failed probably because some timing issues.

**Pull request merged**